src/backends/gr.jl

```diff
@@ -837,7 +842,504 @@ function _update_min_padding!(sp::Subplot{GRBackend})
     if sp[:colorbar_title] != ""
         rightpad += 4mm
     end
-    sp.minpad = Tuple(dpi * [leftpad, toppad, rightpad, bottompad])
+    sp.minpad = Tuple{AbsoluteLength,AbsoluteLength,AbsoluteLength,AbsoluteLength}(dpi * [leftpad, toppad, rightpad, bottompad])
+end
+
+@noinline function viewport_calcs(viewport_plotarea,viewport_size,data_lims,ratio)
```

Comment:
Splitting out functions because the `(function () ... end)()` trick didn't make a noticable change. However, these still cause inference time because it's hit with values that are uninferred, and the problem is largely due to these being a function barrier. @vtjnash should `@nospecialize` do anything about that? I'm kind of confused why it doesn't.